trio/_core/_run.py

```diff
@@ -366,8 +428,7 @@ def _exc_filter(self, exc):
     def _close(self, exc):
         scope_task = current_task()
         if (
-            exc is not None and self.cancel_called
```

Comment:
It looks like `self.cancel_called` is in this if statement to ensure that raising `Cancelled` won't be caught here unless `scope.cancel()` was called. Looking at the logic in `_pending_cancel_scope` it looks like you can't return a scope unless that is already true, so this check is redundant.

I have removed it because there are now cases where you can be cancelled gracefully without having been explicitly cancelled.